README.md

```diff
@@ -130,19 +156,242 @@ Usage: txid2txref [options] <txid>
+Error: transaction f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7521234 not found.
 
-$ # no credentials...will look in <homedir>/.bitcoin/bitcoin.conf
 $ ./src/txid2txref f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107
-txid: f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107
-txref: txtest1-xyv2-xzyq-qqm5-tyke
+{
+  "txid": "f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107",
+  "txref": "txtest1-xyv2-xzyq-qqpq-q6k0a23",
+  "network": "test",
+  "block-height": "1152194",
+  "transaction-position": "1",
+  "utxo-index": "1",
+  "query-string": "f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107"
+}
+```
+
+txid2txref can also output an extended txref if given a utxoIndex > 0, or by using the flag
+`--extended`.
+
+```
+$ ./src/txid2txref f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107 | jq '{txref}'
+{
+  "txref": "txtest1-xyv2-xzyq-qqm5-tyke"
+}
+
+$ ./src/txid2txref --extended f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107 | jq '{txref}'
+{
+  "txref": "txtest1-xyv2-xzyq-qqqq-qc5x2nf"
+}
+
+$ ./src/txid2txref --utxoIndex 1 f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107 | jq '{txref}'
+{
+  "txref": "txtest1-xyv2-xzyq-qqpq-q6k0a23"
+}
+```
+
+Alternately, instead of a txid, txid2txref can be given a txref to
+decode and look up the transaction info in bitcoind:
+
+```
+$ ./src/txid2txref txtest1-xyv2-xzyq-qqpq-q6k0a23
+{
+    "txid":
+    "f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107",
+    "txref": "txtest1-xyv2-xzyq-qqpq-q6k0a23",
+    "network": "test",
+    "block-height": "1152194",
+    "transaction-position": "1",
+    "utxo-index": "1",
+    "query-string": "txtest1-xyv2-xzyq-qqpq-q6k0a23"
+}
+```
+
+
+# Running createBtcrDid
+
+You can ask createBtcrDid for help, and it will show all the runtime
+options available:
+
+```
+> ./createBtcrDid --help
+Usage: createBtcrDid [options] <inputXXX> <changeAddress> <network> <WIF> <fee> <ddoRef>
+
+ -h  --help                 Print this help 
+ --rpchost [rpchost or IP]  RPC host (default: 127.0.0.1) 
+ --rpcuser [user]           RPC user 
+ --rpcpassword [pass]       RPC password 
+ --rpcport [port]           RPC port (default: try both 8332 and 18332) 
+ --config [config_path]     Full pathname to bitcoin.conf (default: <homedir>/.bitcoin/bitcoin.conf) 
+ --utxoIndex [index]        Index # of which UTXO to use from the input transaction (default: 0) 
+
+<inputXXX>      input: (bitcoin address, txid, txref, or txref-ext) needs at least slightly more unspent BTCs than your offered fee
+<outputAddress> output bitcoin address: will receive transaction change and be the basis for your DID
+<network>       the bitcoin network you want to use, 'main' or 'test'
+<WIF>           WIF representation of your private key
+<fee>           fee you are willing to pay (suggestion: >0.001 BTC)
```

Comment:
The fee option should use the estimatefee RPC call, and specify a number of blocks that is acceptable (suggestion: 5).  One bonus of using estimatefee is that it will allow a sensible default for both testnet and mainnet.

https://bitcoin.org/en/developer-reference#estimatefee

Reply:
Great idea. I'll add that.